### PR TITLE
Generate Cartogram using Quadtree-Delaunay Triangulation or qtdt Method, Parallelisation Speedups, and Time Report

### PR DESCRIPTION
**Pull request by nihalzp**
./bin/cartogram ./sample_data/belgium/belgium.geojson ./sample_data/belgium/belgium_population2019.csv -Q

-T flag to output quadtree ps files.

### CODE REVIEW

**Review by mgastner — CHANGES_REQUESTED** [review]
Excellent job! Apart from minor remarks on code comments and variable names, there are two main issues:
- The projection only needs to be pushed back into proj_sequence at the very end of flatten_density(). 
- The grading of the quadtree needs to be turned on. 

I also left a comment about avoiding double-counting of polygon vertices on shared boundaries.

**Review by mgastner — CHANGES_REQUESTED** [review]
When I run `cartogram sample_data/bahamas/bahamas.geojson sample_data/bahamas/bahamas_population2010.csv -pQs`, the following polygon is invalid, apparently because the final point is repeated three times, and then the code exits with an error:

```
303.627 212.143
303.296 211.821
303.641 212.029
303.786 211.997
303.874 212.111
303.874 212.111
303.874 212.111
```

Can you please check whether the new densification code produces repeated points? As an indicator, you can check whether the polygon is simple. It is a strict requirement for CGAL's simplification to work.

**Review by mgastner — COMMENTED** [review]
I believe the variable `found_both_faces` is unnecessary. I also would like some clarification about finding a minimum in parallel. For me as a noobie to omp, it looks as if the omp-for-loop in `InsetState::bbox()` causes a "data race".

File: src/inset_state/inset_state.cpp [hunk 1]
```diff
@@ -19,6 +117,7 @@ Bbox InsetState::bbox() const
   double inset_xmax = -dbl_inf;
   double inset_ymin = dbl_inf;
   double inset_ymax = -dbl_inf;
+#pragma omp parallel for
```
[the remark this answers is not in the document]
Comment (nihalzp) [reply src/inset_state/inset_state.cpp]: Good observation. It could indeed lead to race conditions. I have used reduction to avoid it.

**Comment by mgastner** [comment]
The innovations from this pull request were moved into another pull request #140. That pull request was merged into main.

**Pull request closed without merging**